libsolidity/analysis/ImmutableValidator.cpp

```diff
@@ -104,17 +105,49 @@ bool ImmutableValidator::visit(MemberAccess const& _memberAccess)
 
 bool ImmutableValidator::visit(IfStatement const& _ifStatement)
 {
-	bool prevInBranch = m_inBranch;
+	auto prevInBranch = m_inBranch;
```

Comment:
Why the change away from `bool`?

Reply:
Good point. Reverted it to keep changes minimal